autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/SelectRatesClassification.py

```diff
@@ -0,0 +1,123 @@
+import warnings
+from functools import partial
+from typing import Any, Dict, Optional
+
+from ConfigSpace.conditions import NotEqualsCondition
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformFloatHyperparameter,
+)
+
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.feature_selection import GenericUnivariateSelect, chi2, f_classif, mutual_info_classif
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import FitRequirement, HyperparameterSearchSpace, add_hyperparameter, get_hyperparameter
+
+
+class SelectRatesClassification(autoPyTorchFeaturePreprocessingComponent):
+    """
+    Univariate feature selector by selecting the best features based on
+    univariate statistical tests. Tests can be one of 'chi2, 'f_classif', 'mutual_info_classif'
+    """
+    def __init__(self, alpha: float = 0.1,
+                 score_func: str = "chi2",
+                 mode: str = "fpr",
+                 random_state: Optional[np.random.RandomState] = None
+                 ):
+        self.mode = mode
+        self.alpha = alpha
+        if score_func == "chi2":
+            self.score_func = chi2
+        elif score_func == "f_classif":
+            self.score_func = f_classif
+        elif score_func == "mutual_info_classif":
+            self.score_func = partial(mutual_info_classif,
+                                      random_state=random_state)
+            # mutual info classif constantly crashes without mode percentile
+            self.mode = "percentile"
```

Comment:
Probably we should define something like a class or dict for statistical tests.
Considering the fact that each class takes different choices, I think dict will be the best option.